libr/bin/p/bin_coff.c

```diff
@@ -255,63 +261,171 @@ static RList *libs(RBinFile *bf) {
 	return NULL;
 }
 
-static RList *relocs(RBinFile *bf) {
-	struct r_bin_coff_obj *bin = (struct r_bin_coff_obj*)bf->o->bin_obj;
+#define BYTES_PER_IMP_RELOC		8
+
+static RList *_relocs_list(RBin *rbin, struct r_bin_coff_obj *bin, bool patch, ut64 impmap) {
 	RBinReloc *reloc;
 	struct coff_reloc *rel;
 	int j, i = 0;
 	RList *list_rel;
 	list_rel = r_list_new ();
-	if (!list_rel || !bin || !bin->scn_hdrs) {
+	HtUU *imp_vaddr_ht = NULL;
+	if (patch) {
+		imp_vaddr_ht = ht_uu_new0 ();
+	}
+	if (!list_rel || !bin || !bin->scn_hdrs || (patch && !imp_vaddr_ht)) {
```

Comment:
use r_return_val_if_fail.

Reply:
I think only `bin` can be placed in a `r_return_val_if_fail`. `list_rel` and `imp_vaddr_ht` are allocated at runtime and should not be checked with `r_return_`.